geomesa-hbase/geomesa-hbase-datastore/src/main/java/org/locationtech/geomesa/hbase/filters/JSimpleFeatureFilter.java

```diff
@@ -0,0 +1,102 @@
+package org.locationtech.geomesa.hbase.filters;
+
+import org.apache.hadoop.hbase.Cell;
+import org.apache.hadoop.hbase.CellUtil;
+import org.apache.hadoop.hbase.exceptions.DeserializationException;
+import org.apache.hadoop.hbase.filter.FilterBase;
+import org.apache.hadoop.hbase.util.Bytes;
+import org.geotools.filter.text.cql2.CQLException;
+import org.geotools.filter.text.ecql.ECQL;
+import org.locationtech.geomesa.features.kryo.KryoFeatureSerializer;
+import org.locationtech.geomesa.utils.geotools.SimpleFeatureTypes;
+import org.opengis.feature.simple.SimpleFeature;
+import org.opengis.feature.simple.SimpleFeatureType;
+import org.locationtech.geomesa.features.interop.SerializationOptions;
+
+import java.io.IOException;
+
+public class JSimpleFeatureFilter extends FilterBase {
+    String sftString;
+    SimpleFeatureType sft;
+    KryoFeatureSerializer serializer;
+
+    org.opengis.filter.Filter filter;
+    String filterString;
+
+    public JSimpleFeatureFilter(String sftString, String filterString) {
+        this.sftString = sftString;
+        configureSFT();
+        this.filterString = filterString;
+        configureFilter();
+    }
+
+    public JSimpleFeatureFilter(SimpleFeatureType sft, org.opengis.filter.Filter filter) {
+        this.sft = sft;
+        this.filter = filter;
+        this.sftString = SimpleFeatureTypes.encodeType(sft, true);
+        this.filterString = ECQL.toCQL(filter);
+    }
+
+    private void configureSFT() {
+        sft = SimpleFeatureTypes.createType("QuickStart", sftString);
```

Comment:
you can use an empty string, that's what we do in other iterator implementations.